src/app/digital/models/ioobjects/inputs/Clock.ts

```diff
@@ -33,6 +35,13 @@ export class Clock extends DigitalComponent {
     public activate(bool: boolean): void {
         super.activate(bool);
         this.isOn = bool;
+
+        // If WiringTool is active (and its parent is a clock), activate its wire as well
+        if (WiringTool.getWire() !== undefined && WiringTool.getWire().getP1Component().getName() == "Clock") {
+            const w = WiringTool.getWire() as DigitalWire;
+            w.activate(bool);
+        }
+
```

Comment:
So as I mentioned in class, the idea of importing something like the `WiringTool` into something like the `Clock` tends to be a bad idea. (and also again don't feel bad, it's super awesome that you were able to solve the issue regardless and I never mentioned that this is "bad" practice).

The reason I would say this is "bad" is because the Clock is an isolated component that really shouldn't know anything about the WiringTool's existence, and I can imagine a scenario where the Clock exists despite the WiringTool not existing like inside an IC or something like that.

But anyways –– I think the solution here is to have the wire be drawn as On/Off completely depending on whether the wire's `getInput().getIsOn()` and so the solution would be located probably in `src/app/digital/rendering/ToolRenderer` where you should be able to find the rendering for the `WiringTool`

Reply:
Yeah, I realized when you mentioned that in class the other day that my solution was definitely not the best.
I was working on it a bit more and I thought I found another solution, but it was causing errors when I was running the yarn tests, so I tried yours and was able to get everything working with only one line change in WireRenderer (plus what I had done in DigitalWire).

I'm just confused about one thing, I noticed it when I was testing my other solution as well:
<img width="955" alt="opencirc" src="https://user-images.githubusercontent.com/65261267/140975932-6249adac-f8d5-4ba7-b017-9d738fbfdeb7.png">
<img width="955" alt="opencirc2" src="https://user-images.githubusercontent.com/65261267/140976297-5518d0f6-5013-4c03-be6c-bccf4754dd57.png">
Is it expected that not gates, nands, nors, etc. power wires without a source?
Besides that everything seems good, I just wanted to make sure it's expected before I commit.

Reply:
Awesome! And yeah that should be totally fine for NOTGate's and such